src/VirtualMonoRepo/source-mappings.json

```diff
@@ -210,11 +748,30 @@
         },
         {
             "name": "winforms",
-            "defaultRemote": "https://github.com/dotnet/winforms"
+            "defaultRemote": "https://github.com/dotnet/winforms",
+            "include": [
+                "src/System.Windows.Forms/tests/IntegrationTests/MauiTests/lib/GenStrings.dll",
+                "src/System.Windows.Forms/tests/IntegrationTests/MauiTests/lib/WinformsTestLib.dll"
+            ],
+            "exclude": [
+                // https://github.com/dotnet/source-build/issues/3745
```

Comment:
Is excluding these files going to break WinForms and WPF product builds?

Reply:
They should be removed by [the script to remove binaries.](https://github.com/dotnet/source-build/issues/4089) @mthalman - do you know if it will be alright to leave these uncloaked if they're going to be removed for SB with the script?

Reply:
What is the script to remove binaries going to look like? Is it just going to delete all files with [these extensions](https://github.com/dotnet/installer/pull/18563/files#diff-9d9a231b136610b76071237ea260d7913cdbc4eef6e9371ce038a7f41d11b273R41-R48)?

If it is the case, I think the scheme can be simplified to:
- Delete exclusion of the binary file extensions in source-mappings.json
- Delete inclusion of the additional files added in this PR
- The only thing left in source-mappings.json is list of repos and exclusion of files with problematic license like src/package/licenses/LICENSE_VS.txt

Would that work?

Reply:
We want to be checking that the binaries are not getting added to the contributing repos in the first place. This change does not prevent that. The VMR mapping script is going to trim the binary.

In the worst case, it is going to lead to broken shipping product bits to be produced by the VMR build. 

In a better case, it is going to lead to (hard-to-understand) build break.

I think the check for new binaries should be de-coupled from the VMR mapping script. Ideally, the check for the new binaries would be part of the source-build leg in the contributing repos.

Reply:
I agree that the scans should get lowered to the SB legs because there will be no installer-based VMR sync in the future to safe-guard against this. That would require the information about binaries to be present in each repo which also is not a bad idea.

Reply:
> There are repos which we do not control though and we won't be able to put this information in those.

Right, the check can run for the whole VMR as well with the additional allow list. That's fine.

My main point is that the binary allow list check should be de-coupled from the VMR mapping script.

Reply:
For my understanding, I think what @jkotas is saying is that we should treat source-mappings as permanent mappings, with no exclusions except permanent ones that both MSFT and distro maintainers don't want (licenses).

Then we have a separate list of files/patterns that should be deleted with a provided script. Something easily parseable in bash. This should **not** be in source-mappings.json. It's just located in each repo, and if that repo has submodules, then the exclusion file 'speaks' for the inlined submodules. The binary check keys off this in repo legs, and the exclusion script just reads patterns from the file to delete.

Reply:
I like the direction this is heading.

Reply:
... as it is achieving both goals - maintainability + raising the bar for adding new binaries.

Reply:
I agree that this feels like the right direction. I've updated the PR in https://github.com/dotnet/installer/pull/18563/commits/4a7d5356a08f4765d4e9b85362fa78889e276af4 to remove all binary cloaks except for the necessary exclusions. Ran [license scanning with these changes as well](https://dev.azure.com/dnceng/internal/_build/results?buildId=2376360&view=logs&j=ab6b9953-d92d-5494-251d-2f65ef31076b).